src/legacy/server/saved_objects/lib/export.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Boom from 'boom';
+import { SavedObject, SavedObjectsClient } from '../service/saved_objects_client';
+
+interface ObjectToExport {
+  id: string;
+  type: string;
+}
+
+interface ExportObjectsOptions {
+  types?: string[];
+  objects?: ObjectToExport[];
+  savedObjectsClient: SavedObjectsClient;
+  exportSizeLimit: number;
+}
+
+export async function getSortedObjectsForExport({
+  types,
+  objects,
+  savedObjectsClient,
+  exportSizeLimit,
+}: ExportObjectsOptions) {
+  let objectsToExport: SavedObject[] = [];
+  if (objects) {
+    if (objects.length > exportSizeLimit) {
+      throw Boom.badRequest(`Can't export more than ${exportSizeLimit} objects`);
+    }
+    ({ saved_objects: objectsToExport } = await savedObjectsClient.bulkGet(objects));
+    objectsToExport = objectsToExport.filter(obj => !obj.error);
```

Comment:
What to do when objects aren't found or failed to be fetched?

Reply:
I think when specific objects are requested, and any of them are missing or failed to be fetched, we should fail and do our best to combine the errors, or return the errors in an array within the response.